### PR TITLE
refact(provider/kubernetes): refactor delete pods operation

### PR DESCRIPTION
**Pull request by chlung**
About this PR,

- Use pod deletion not resize for recreating and destroying statefulset and daemonset operations.
- Prevent resize daemonset operation. 

Thanks.
We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy [hunk 1]
```diff
@@ -172,7 +172,7 @@ class KubernetesClientApiAdapter {
         }
 
         return list.items
-      } catch (ApiException ex) {
+      } catch (ApiException e) {
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: Can you pull this fix into a separate PR that we can merge into the release-1.5.x branch as well?

Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/api/KubernetesClientApiAdapter.groovy]: @lwander will do it.

**Comment by chlung** [comment]
@lwander please review.  thanks.
